{
    "url": "https://api.github.com/repos/spack/spack/issues/6965",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/6965/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/6965/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/6965/events",
    "html_url": "https://github.com/spack/spack/issues/6965",
    "id": 289241487,
    "node_id": "MDU6SXNzdWUyODkyNDE0ODc=",
    "number": 6965,
    "title": "glib not building on RedHat 7 with dtrace",
    "user": {
        "login": "luigi-calori",
        "id": 5165864,
        "node_id": "MDQ6VXNlcjUxNjU4NjQ=",
        "avatar_url": "https://avatars.githubusercontent.com/u/5165864?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/luigi-calori",
        "html_url": "https://github.com/luigi-calori",
        "followers_url": "https://api.github.com/users/luigi-calori/followers",
        "following_url": "https://api.github.com/users/luigi-calori/following{/other_user}",
        "gists_url": "https://api.github.com/users/luigi-calori/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/luigi-calori/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/luigi-calori/subscriptions",
        "organizations_url": "https://api.github.com/users/luigi-calori/orgs",
        "repos_url": "https://api.github.com/users/luigi-calori/repos",
        "events_url": "https://api.github.com/users/luigi-calori/events{/privacy}",
        "received_events_url": "https://api.github.com/users/luigi-calori/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 446616547,
            "node_id": "MDU6TGFiZWw0NDY2MTY1NDc=",
            "url": "https://api.github.com/repos/spack/spack/labels/build-error",
            "name": "build-error",
            "color": "ff3300",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 2,
    "created_at": "2018-01-17T12:00:41Z",
    "updated_at": "2018-03-19T05:38:43Z",
    "closed_at": "2018-03-19T05:38:43Z",
    "author_association": "CONTRIBUTOR",
    "active_lock_reason": null,
    "body": "On my redhat 7 that has dtrace installed, spack fail to build glib, this likely relates to PYTHONHOME being defined when depending on python, this makes system python scripts that sbang to system python exit with error  \r\nImportError: No module named site\r\n\r\n-----\r\n\r\nSummary\r\n\r\n### Expected Result\r\nspack install glib\r\n\r\n### Actual Result\r\n\r\n==> Applied patch no-Werror=format-security.patch\r\n==> Building glib [AutotoolsPackage]\r\n==> Executing phase: 'autoreconf'\r\n==> Executing phase: 'configure'\r\n==> Executing phase: 'build'\r\n==> Error: ProcessError: Command exited with status 2:\r\n    'make' '-j48'\r\n\r\n4 errors found in build log:\r\n     [ ... ]\r\n     562   Making all in glib\r\n     563   make[2]: Entering directory `/dev/shm/cibo19/spack-stage/spack-stage-giUdzE/glib-2.55.1/glib'\r\n     564     GEN      glibconfig-stamp\r\n     565     GEN      glib_probes.h\r\n     566     GEN      glib_probes.lo\r\n     567   ImportError: No module named site\r\n  >> 568   make[2]: *** [glib_probes.h] Error 1\r\n     569   make[2]: *** Waiting for unfinished jobs....\r\n     570   ImportError: No module named site\r\n  >> 571   make[2]: *** [glib_probes.lo] Error 1\r\n\r\n### Steps to reproduce the issue\r\n\r\n```console\r\n$ which dtrace\r\n$ head $(which dtrace)\r\n$ dtrace\r\nUsage /usr/bin/dtrace [--help] [-h | -G] [-C [-I<Path>]] -s File.d [-o <File>]   ### OK\r\n$ spack install --keep-stage glib@2.55.1%gcc@4.8.5\r\n$ spack env glib@2.55.1%gcc@4.8.5 dtrace\r\nImportError: No module named site   ### python error\r\n$ spack env glib@2.55.1%gcc@4.8.5 /bin/bash\r\n$ dtrace\r\nImportError: No module named site ### python error\r\n$ unset PYTHONHOME\r\n$ dtrace\r\nUsage /usr/bin/dtrace [--help] [-h | -G] [-C [-I<Path>]] -s File.d [-o <File>]   ### OK\r\n\r\n\r\n",
    "performed_via_github_app": null
}